lib/app/features/nostr/providers/nostr_notifier.dart

```diff
@@ -13,23 +17,95 @@ class NostrNotifier extends _$NostrNotifier {
   @override
   FutureOr<void> build() {}
 
-  Future<void> sendEvents(List<EventMessage> events, {String? pubkey}) async {
-    final userRelays = pubkey != null
-        ? (await ref.read(userRelaysProvider(pubkey).future))
-        : (await ref.read(currentUserRelaysProvider.future));
+  Future<void> send(
+    List<EventMessage> events, {
+    ActionSource actionSource = const ActionSourceCurrentUser(),
+  }) async {
+    final relay = await _getRelay(actionSource);
+    // await relay.sendEvents(events);
+    // TODO: uncomment when our relays are used
+    await Future.wait(events.map(relay.sendEvent).toList());
+  }
 
-    if (userRelays == null) {
-      throw Exception('User relays are null');
-    }
+  Future<void> sendOne(
+    EventMessage event, {
+    ActionSource actionSource = const ActionSourceCurrentUser(),
+  }) async {
+    return send([event], actionSource: actionSource);
+  }
 
-    final relay = await ref.read(relayProvider(userRelays.list.random.url).future);
+  Future<Stream<EventMessage>> request(
```

Comment:
need to return either `Future` or `Stream`, but not `Future<Stream>`

Reply:
So it can be like this:

```
Stream<EventMessage> request(
    RequestMessage requestMessage, {
    ActionSource actionSource = const ActionSourceCurrentUser(),
  }) async* {
      final relay = await _getRelay(actionSource);
      yield* requestEvents(requestMessage, relay);
```